src/tile_setter.rs

```diff
@@ -0,0 +1,213 @@
+use crate::{lib::*, tile::Tile};
+
+/// A tool used to set multiple tiles at a time.
+///
+/// This is the preferred and fastest way to set tiles. Optionally, you can set them individually.
+#[derive(Clone, Debug, Default)]
+pub struct TileSetter(Vec<(Vec3, Tile, usize)>);
+
+impl TileSetter {
+    /// Constructs a new `TileSetter` with the type `Tile`.
+    ///
+    /// This is effectively a wrapped vector with a tuple of a coordinate and
+    /// `Tile`. By itself it does not set the tiles but it is used as a helper
+    /// to keep data clear and concise.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    ///
+    /// let mut setter = TileSetter::new();
+    /// ```
+    pub fn new() -> TileSetter {
```

Comment:
https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=expected#types-eagerly-implement-common-traits-c-common-traits
> Note that it is common and expected for types to implement both Default and an empty new constructor. new is the constructor convention in Rust, and users expect it to exist, so if it is reasonable for the basic constructor to take no arguments, then it should, even if it is functionally identical to default.

cart has a different opinion however https://github.com/bevyengine/bevy/pull/765#discussion_r520960497
> This is one of the few places where my opinions diverge from the api guidelines. I prefer "default only" because:
> 
> * it means theres "one way" to instantiate the type (a good thing ... lowers cognitive load)
> * less code (easier to read, easier to maintain, _marginally_ faster to compile)

I think I like `new()` because as you add features, you may eventually need more than zero arguments for a constructor.

Having said that, `bevy` takes advantage of `Default` a lot, e.g. for initialising new resources (`Local` or otherwise).

Reply:
Yeah, I think I agree with cart here but I don't have super strong feelings. Having two zero argument constructors seems ambiguous to me, if there's just one there's no question and there's only one code path.

Not having a zero argument `new` now doesn't prevents you from adding one later.

Reply:
That and a lot of the time new = default. But I really disagree with have pub types as Bevy does and construct with defaults.

Reply:
Resolved now as TileSetter is deprected.